src/hooks/useBalance.ts

```diff
@@ -0,0 +1,54 @@
+import { ApiPromise } from '@polkadot/api';
+import { WeightV2 } from '@polkadot/types/interfaces';
+import { FrameSystemAccountInfo } from '@polkadot/types/lookup';
+import { UseQueryResult } from '@tanstack/react-query';
+import { useMemo } from 'preact/compat';
+import { cacheKeys, inactiveOptions, QueryOptions } from '../constants/cache';
+import { mockERC20 } from '../contracts/MockERC20';
+import { useGlobalState } from '../GlobalStateProvider';
+import { nativeToDecimal, prettyNumbers } from '../helpers/parseNumbers';
+import { useNodeInfoState } from '../NodeInfoProvider';
+import { useContract } from './useContract';
+
+export type UseBalanceResponse = UseQueryResult<FrameSystemAccountInfo | undefined, unknown> & {
+  balance?: number;
+  formatted?: string;
+};
+
+export const createOptions = (api: ApiPromise) => ({
+  gasLimit: api.createType('WeightV2', {
+    refTime: '100000000000',
+    proofSize: '1000000',
+  }) as WeightV2,
+  storageDepositLimit: null,
+});
+
+export const useBalance = (tokenAddress?: string, options?: QueryOptions): UseBalanceResponse => {
+  const {
+    state: { api },
+  } = useNodeInfoState();
+  const { address } = useGlobalState().walletAccount || {};
```

Comment:
Maybe, in this case we should actually consider falling back to some hard-coded account address. As far as I know, for querying data the account doesn't have to actually 'exist' on the network or rather, actually have some tokens.

With the current implementation, the user would only see some balances loading once he connects his wallet account. But as a user, I would expect to already be able to interact with the Swap UI on a basic level without needing to connect my account.

Reply:
So I would suggest we define a constant 'FALLBACK_QUERY_SUBSTRATE_ACCOUNT' or similar, that is a string holding any account address. 
We could use '5GjwVRkXVyFjZxHfbGWNP3ed44pjvwFes8AJzniMzznWa27i', which is the address of the joint product-tech account.

Reply:
Maybe I am misunderstanding or maybe the name of the hook needs to be more descriptive. This should be used for fetching user/wallet balances (his tokens, deposits...). So maybe renaming the hook?

Reply:
Ah true, it's always just about the user balance. Nevermind then, we wouldn't need the fallback for that.